source/extensions/filters/network/zookeeper_proxy/filter.cc

```diff
@@ -19,55 +21,170 @@ namespace NetworkFilters {
+  initOpCode(OpCodes::SetWatches, stats_.setwatches_resp_, stats_.setwatches_resp_fast_,
+             stats_.setwatches_resp_slow_, "setwatches_resp");
+  initOpCode(OpCodes::SetWatches2, stats_.setwatches2_resp_, stats_.setwatches2_resp_fast_,
+             stats_.setwatches2_resp_slow_, "setwatches2_resp");
+  initOpCode(OpCodes::CheckWatches, stats_.checkwatches_resp_, stats_.checkwatches_resp_fast_,
+             stats_.checkwatches_resp_slow_, "checkwatches_resp");
+  initOpCode(OpCodes::RemoveWatches, stats_.removewatches_resp_, stats_.removewatches_resp_fast_,
+             stats_.removewatches_resp_slow_, "removewatches_resp");
+  initOpCode(OpCodes::GetEphemerals, stats_.getephemerals_resp_, stats_.getephemerals_resp_fast_,
+             stats_.getephemerals_resp_slow_, "getephemerals_resp");
   initOpCode(OpCodes::GetAllChildrenNumber, stats_.getallchildrennumber_resp_,
+             stats_.getallchildrennumber_resp_fast_, stats_.getallchildrennumber_resp_slow_,
              "getallchildrennumber_resp");
-  initOpCode(OpCodes::Close, stats_.close_resp_, "close_resp");
+  initOpCode(OpCodes::Close, stats_.close_resp_, stats_.close_resp_fast_, stats_.close_resp_slow_,
+             "close_resp");
 }
 
-void ZooKeeperFilterConfig::initOpCode(OpCodes opcode, Stats::Counter& counter,
-                                       absl::string_view name) {
+ErrorBudgetResponseType
+ZooKeeperFilterConfig::errorBudgetDecision(const OpCodes opcode,
+                                           const std::chrono::milliseconds& latency) {
+  if (latency_threshold_map_.empty()) {
+    return ErrorBudgetResponseType::NONE;
+  }
+
+  // Set latency threshold for the current opcode.
+  std::chrono::milliseconds latency_threshold = default_latency_threshold_;
+  int32_t opcode_val = enumToSignedInt(opcode);
+  auto it = latency_threshold_map_.find(opcode_val);
+  if (it != latency_threshold_map_.end()) {
+    latency_threshold = it->second;
+  }
+
+  // Determine fast/slow response based on the threshold.
+  if (latency <= latency_threshold) {
+    return ErrorBudgetResponseType::FAST;
+  }
+
+  return ErrorBudgetResponseType::SLOW;
+}
+
+void ZooKeeperFilterConfig::initOpCode(OpCodes opcode, Stats::Counter& resp_counter,
+                                       Stats::Counter& resp_fast_counter,
+                                       Stats::Counter& resp_slow_counter, absl::string_view name) {
   OpCodeInfo& opcode_info = op_code_map_[opcode];
-  opcode_info.counter_ = &counter;
+  opcode_info.resp_counter_ = &resp_counter;
+  opcode_info.resp_fast_counter_ = &resp_fast_counter;
+  opcode_info.resp_slow_counter_ = &resp_slow_counter;
   opcode_info.opname_ = std::string(name);
   opcode_info.latency_name_ = stat_name_set_->add(absl::StrCat(name, "_latency"));
 }
 
+int32_t ZooKeeperFilterConfig::getOpCodeIndex(LatencyThreshold_Opcode opcode) {
+  static absl::flat_hash_map<LatencyThreshold_Opcode, int32_t> opcode_map = {
```

Comment:
I did not use `constexpr` here because the compiler says `constexpr variable cannot have non-literal type 'const absl::flat_hash_map<LatencyThreshold_Opcode, int32_t>'`